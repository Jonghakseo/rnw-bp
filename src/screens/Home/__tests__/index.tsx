import React from "react";
import { Home } from "..";
import renderer from "react-test-renderer";
import { mockNavigation } from "../../../mocks/Setup";
import withProvider from "../../../testHelper";

it("should render correctly", () => {
  const component = renderer.create(withProvider(<Home />));
  expect(component.toJSON()).toMatchSnapshot();
});

it("navigates to details", () => {
  const component = renderer.create(withProvider(<Home />));
  const button = component.root.findByProps({ testID: "go to detail button" });
  button.instance.props.onPress();
  expect(mockNavigation.navigate).toBeCalledWith("Details", { data: "ðŸ¤ª" });
});
