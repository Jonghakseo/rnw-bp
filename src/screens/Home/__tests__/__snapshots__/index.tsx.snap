// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d47271",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text>
    Home Screen
  </Text>
  <Text>
    http://api.example.com
  </Text>
  <Text>
    dev
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="go to detail button"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go to Details
      </Text>
    </View>
  </View>
  <Text>
    
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="get token test"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        get token
      </Text>
    </View>
  </View>
</View>
`;
